pkg/controller/machine/machine_controller.go

```diff
@@ -305,3 +341,35 @@ func (r *ReconcileMachine) deleteNode(ctx context.Context, cluster *clusterv1.Cl
 
 	return corev1Remote.Nodes().Delete(name, &metav1.DeleteOptions{})
 }
+
+// getMachines returns all of the Machine objects that belong to the same
+// cluster as the provided Machine
+func (r *ReconcileMachine) getMachines(ctx context.Context, machine *clusterv1.Machine) ([]*clusterv1.Machine, error) {
+	clusterName := machine.Labels[clusterv1.MachineClusterLabelName]
+	if clusterName == "" {
+		klog.Infof("Machine %q in namespace %q doesn't specify %q label, assuming nil cluster", machine.Name, machine.Namespace, clusterv1.MachineClusterLabelName)
+		return nil, nil
+	}
+
+	mlist := &clusterv1.MachineList{}
+	listOptions := &client.ListOptions{
+		Namespace: machine.Namespace,
+		LabelSelector: labels.Set(map[string]string{
+			clusterv1.MachineClusterLabelName: clusterName,
+		}).AsSelector(),
+	}
+
+	if err := r.Client.List(ctx, listOptions, mlist); err != nil {
+		if apierrors.IsNotFound(err) {
```

Comment:
Does `List` return IsNotFound if there are no machines?

Reply:
client-go does not (it returns an empty list.Items). I don't know about controller-runtime's client, but I would expect an empty list instead of not-found.

Reply:
I just did this in case. I wasn't sure, so this seemed like an okay check.